twake/backend/node/src/core/platform/framework/api/crud-service.ts

```diff
@@ -43,12 +30,11 @@ export class DeleteResult<Entity> extends EntityTarget<Entity>  {
 
 export declare type EntityId = string | number;
 
-export declare type EntityOperationResult<Entity> = CreateResult<Entity> | UpdateResult<Entity> | DeleteResult<Entity>;
+export declare type EntityOperationResult<Entity> = SaveResult<Entity> | DeleteResult<Entity>;
 
 export interface CRUDService<Entity> {
-  create(item: Entity): Promise<CreateResult<Entity>>;
-  get(id: EntityId): Promise<Entity>;
-  update(id: EntityId, item: Entity, /* TODO: Options */): Promise<UpdateResult<Entity>>;
-  delete(id: EntityId): Promise<DeleteResult<Entity>>;
-  list(/* TODO: Options */): Promise<Entity[]>;
+  save(item: Entity): Promise<SaveResult<Entity>>;
```

Comment:
OK for the primary key, but it must be a generic so that we can choose it based on the type of database:

```js
export interface CRUDService<Entity, PrimaryKey> {
   get(primaryKey: PrimaryKey): Promise<Entity>;
}
```

Reply:
Did you have time to work on this? If not I will do it next week, focus on other parts